aesd-char-driver/main.c

```diff
@@ -218,12 +219,170 @@ ssize_t aesd_write(struct file *filp, const char __user *buf, size_t count,
 
     return retval;
 }
+
+loff_t aesd_llseek(struct file *filp, loff_t offset, int whence)
+{
+    loff_t retval;
+    int status;
+
+    struct aesd_dev *dev = filp->private_data;
+
+    // Locking using the interruptable version so that it can handle signals unlike non interruptable versions
+    status = mutex_lock_interruptible(&dev->lock);
+    if (status != 0)
+    {
+        PDEBUG("Error: mutex_lock_interruptible failed");
+        retval = -ERESTARTSYS;
+        goto out;
+    }
+
+    // Calling the system call fixed_size_llseek
+    retval = fixed_size_llseek(filp, offset, whence, dev->aesd_circular_buffer.size_cb);
```

Comment:
@nihalthirunakarasu how could you make it so `fixed_size_llseek` does not need to be in your locking interval?